backend/src/main/java/com/huellapositiva/application/dto/ChangePasswordDto.java

```diff
@@ -0,0 +1,28 @@
+package com.huellapositiva.application.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+@Data
+@Builder
+@AllArgsConstructor
+public class ChangePasswordDto {
+
+    @NotEmpty(message = "You must to write something")
+    @NotBlank(message = "The password is not blank space")
+    @Size(min = 6, message = "The password is too short")
+    @Pattern(regexp = "^[a-zA-Z0-9àèìòùÀÈÌÒÙáéíóúýÁÉÍÓÚÝâêîôûÂÊÎÔÛãñõÃÑÕäëïöüÿÄËÏÖÜŸ.,:;+*/|%!@#$&€=<>() -]*$", message = "The new password does not match with the regular expresion")
+    private final String newPassword;
+
+    @NotEmpty(message = "You must to write something")
+    @NotBlank(message = "The password is not blank space")
+    @Size(min = 6, message = "The password is too short")
+    @Pattern(regexp = "^[a-zA-Z0-9àèìòùÀÈÌÒÙáéíóúýÁÉÍÓÚÝâêîôûÂÊÎÔÛãñõÃÑÕäëïöüÿÄËÏÖÜŸ.,:;+*/|%!@#$&€=<>() -]*$", message = "The old password does not match with the regular expresion")
```

Comment:
I would not check this in the "oldPassword" property. You never know what the previously validation rules were and it might not be backwards compatible.

Image in the future we decide that the minimum length should be 5, and some time later that it should be 4. How would you distinguish that the user trying to update their password had one restriction or the other?

You should only validate the new password upon the current rules.

Reply:
The old password does not check size and pattern anymore.